src/main/java/fhirspark/adapter/FollowUpAdapter.java

```diff
@@ -0,0 +1,274 @@
+            coding.setSystem("http://hl7.org/fhir/ValueSet/reason-medication-status-codes");
+            coding.setCode("395009001");
+            coding.setDisplay("Medication stopped - side effect");
+            cc.addCoding(coding);
+            medicationStatement.addStatusReason(cc);
+        }
+
+        if (!followUp.getId().startsWith("followUp_" + patientId + "_")) {
+            throw new IllegalArgumentException("Invalid followUp id!");
+        }
+
+        //medicationStatement.setIssued(new Date(Long.valueOf(mtb.getId().replace("mtb_" + patientId + "_", ""))));
+
+        if (followUp.getResponse() != null) {
+            ResponseCriteria response = followUp.getResponse();
+            ArrayList<String> responseTags = new ArrayList<String>();
+            ArrayList<Boolean> responseValues = new ArrayList<Boolean>();
+            final int respCount = 12;
+
+            responseTags.add("Pd3");
+            responseValues.add(response.getPd3());
+            responseTags.add("Pr3");
+            responseValues.add(response.getPr3());
+            responseTags.add("Cr3");
+            responseValues.add(response.getCr3());
+            responseTags.add("Sd3");
+            responseValues.add(response.getSd3());
+            responseTags.add("Pd6");
+            responseValues.add(response.getPd6());
+            responseTags.add("Pr6");
+            responseValues.add(response.getPr6());
+            responseTags.add("Cr6");
+            responseValues.add(response.getCr6());
+            responseTags.add("Sd6");
+            responseValues.add(response.getSd6());
+            responseTags.add("Pd12");
+            responseValues.add(response.getPd12());
+            responseTags.add("Pr12");
+            responseValues.add(response.getPr12());
+            responseTags.add("Cr12");
+            responseValues.add(response.getCr12());
+            responseTags.add("Sd12");
+            responseValues.add(response.getSd12());
+
+            for (int i = 0; i < respCount; i++) {
+
+                Observation responseObs = new Observation();
+
+                responseObs.getCode()
+                    .addCoding(GenomicsReportingEnum.THERAPEUTIC_IMPLICATION_CODING.toCoding());
+
+                Annotation note = new Annotation();
+                note.setText(responseTags.get(i));
+                ArrayList<Annotation> notes = new ArrayList<Annotation>();
+                notes.add(note);
+                responseObs.setNote(notes);
+
+                BooleanType bType = new BooleanType(responseValues.get(i));
+                responseObs.setValue(bType);
+
+                DateTimeType dTime = new DateTimeType(followUp.getDate());
+                responseObs.setEffective(dTime);
+
+                responseObs.setId(IdType.newRandomUuid());
+                responseObs.setStatus(Observation.ObservationStatus.FINAL);
+                responseObs.addIdentifier().setSystem("https://cbioportal.org/followUpResponse/")
```

Comment:
I'd like to see this moved to an enum or at least to a static constant of the class.